### PR TITLE
[IA-4010] add deleteAll exception and test

### PR DESCRIPTION
**Pull request by jdcanas**
Adds an exception if you call deleteAll and there is a non-deletable app

### CODE REVIEW

**Comment by jdcanas** [comment]
jenkins retest

**Review by n8landolt — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 1]
```diff
@@ -1227,12 +1233,10 @@ case class AppCannotBeDeletedByWorkspaceIdException(workspaceId: WorkspaceId,
       traceId = Some(traceId)
     )
 
-case class AppCannotBeCreatedException(googleProject: GoogleProject,
```
Comment (lucymcnatt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:1230]: why is this exception class being removed?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala [hunk 2]
```diff
@@ -1227,12 +1233,10 @@ case class AppCannotBeDeletedByWorkspaceIdException(workspaceId: WorkspaceId,
...
-                                       appName: AppName,
-                                       status: AppStatus,
-                                       traceId: TraceId
-) extends LeoException(
-      s"App ${googleProject.value}/${appName.value} cannot be created in ${status} status.",
+case class DeleteAllAppsCannotBePerformed(workspaceId: WorkspaceId, apps: List[App], traceId: TraceId)
+    extends LeoException(
+      s"App(s) in workspace ${workspaceId.value.toString} with (name(s), status(es)) ${apps
+          .map(app => s"(${app.appName.value},${app.status})")} cannot be deleted their status(es).",
```
Comment (lucymcnatt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala]: nit: should this be "cannot be deleted due to their status(es)"?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala (hunk 1 above)
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala:1230]: Its not used anywhere, just cleaning up

**Review by lucymcnatt — APPROVED** [review]

**Pull request merged**